File: python/practices/string/jewels-and-stones.py
def numJewelsInStones(self, J, S):
    return sum(map(J.count, S))


def _numJewelsInStones(self, J, S):
    return sum(map(S.count, J))


def __numJewelsInStones(J, S):
    return sum(s in J for s in S)

File: java/src/main/java/cn/az/code/practices/string/JewelsAndStones.java
package cn.az.code.practices.string;

import java.util.HashSet;
import java.util.Set;

/**
 * @author az
 * @since 2020-06-18 22:07
 */
public class JewelsAndStones {

    public int numJewelsInStones(String J, String S) {
        Set<Character> set = new HashSet<>();
        for (char ch : J.toCharArray()) {
            set.add(ch);
        }
        int count = 0;
        for (char c : S.toCharArray()) {
            if (set.contains(c)) {
                count++;
            }
        }
        return count;
    }

    public int _newJewelInStones(String J, String S) {
        return S.replaceAll("[^" + J + "]", "").length();
    }
}
